### PR TITLE
Cast TokenValue.__str__ return value to string

### LINKED ISSUE #69: Parser fails with 'TypeError' on invalid query "~]"
There's a query from pair of symbols which breaks the error handling of the parser, so we can't get relevant information about the syntax error.

**How to reproduce**:

```python
from luqum.parser import parser

parser.parse('~]')
```

**Expected result**: `ParseSyntaxError`
**Actual result**: `TypeError: __str__ returned non-string (type NoneType)`

**Possible fix**: 

Update the `TokenValue.__str__` [method](https://github.com/jurismarches/luqum/blob/master/luqum/head_tail.py#L21) to always return string. I.e.

```python
class TokenValue:
# ...
    def __str__(self):
        return str(self.value)
```

### PR DESCRIPTION
**Pull request by delkopiso**
fixes #69

### CODE REVIEW

**Review by alexgarel — APPROVED** [review]
Really cool change.

I approve so you can merge as is, but if you find the time to add a test that would be super-cool :-)

File: luqum/head_tail.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def __repr__(self):
         return "TokenValue(%s)" % self.value
 
     def __str__(self):
-        return self.value
+        return str(self.value)
```
Comment (alexgarel) [inline luqum/head_tail.py:21]: Hum sorry I think this would be better:

```suggestion
        return str(self.value) if self.value else ""
```

**Review by alexgarel — CHANGES_REQUESTED** [review]
Sorry my first approval was a bit too quick. Can you consider my suggestion ?

**Comment by delkopiso** [comment]
@alexgarel that makes sense, I've applied your suggestion.

**Review by alexgarel — APPROVED** [review]
Perfect, thank you so much :-)

**Pull request merged**